molfeat/trans/graph/adj.py

```diff
@@ -727,6 +730,7 @@ def transform(self, mols: List[Union[dm.Mol, str]], **kwargs):
 
     def get_collate_fn(
         self,
+        dataset: Union[Dataset, Sequence[BaseData], DatasetAdapter],
```

Comment:
At first glance, I am convinced, this is likely going to fail if the package is not installed (which is the default behaviour with pip).

Please use something like:

```python
from typing import TYPE_CHECKING, Union, Sequence

if requires.check("torch_geometric"):
    from torch_geometric.data import Data
    from torch_geometric.loader.dataloader import Collater
    if TYPE_CHECKING:
         from torch_geometric.data import Dataset # should probably import as PygDataset
         from torch_geometric.data.data import BaseData
         from torch_geometric.data.datapipes import DatasetAdapter
    else:
         Dataset, BaseData, DatasetAdapter = None, None, None
         
```

Then use strings for the typing annotations. e.g `Sequence["BaseData"]`